docs/nomad/README.md

```diff
@@ -0,0 +1,262 @@
+
+    constraint {
+      distinct_hosts = true
+    }
+
+    update {
+      max_parallel     = 1
+      canary           = 1
+      min_healthy_time = "10s"
+      healthy_deadline = "1m"
+      auto_revert      = true
+      auto_promote     = true
+    }
+
+    task "plugin" {
+      driver = "docker"
+
+      config {
+        # Get the latest version on https://hub.docker.com/r/hetznercloud/hcloud-csi-driver/tags
+        image   = "hetznercloud/hcloud-csi-driver:v2.3.2"
+        command = "bin/hcloud-csi-driver-controller"
+      }
+
+      env {
+        CSI_ENDPOINT   = "unix://csi/csi.sock"
+        ENABLE_METRICS = true
+      }
+
+      template {
+        data        = <<EOH
+HCLOUD_TOKEN="{{ with nomadVar "secrets/hcloud" }}{{ .hcloud_token }}{{ end }}"
+EOH
+        destination = "${NOMAD_SECRETS_DIR}/hcloud-token.env"
+        env         = true
+      }
+
+      csi_plugin {
+        id        = "csi.hetzner.cloud"
+        type      = "controller"
+        mount_dir = "/csi"
+      }
+
+      resources {
+        cpu    = 100
+        memory = 64
+      }
+    }
+  }
+}
+```
+
+4. Create a CSI Node Job:
+
+```hcl
+# file: hcloud-csi-node.hcl
+job "hcloud-csi-node" {
+  datacenters = ["dc1"]
+  namespace   = "default"
+  type        = "system"
+
+  group "node" {
+    task "plugin" {
+      driver = "docker"
+
+      config {
+        # Get the latest version on https://hub.docker.com/r/hetznercloud/hcloud-csi-driver/tags
+        image      = "hetznercloud/hcloud-csi-driver:v2.3.2"
+        command    = "bin/hcloud-csi-driver-node"
+        privileged = true
+      }
+
+      env {
+        CSI_ENDPOINT   = "unix://csi/csi.sock"
+        ENABLE_METRICS = true
+      }
+
+      template {
+        data        = <<EOH
+HCLOUD_TOKEN="{{ with nomadVar "secrets/hcloud" }}{{ .hcloud_token }}{{ end }}"
+EOH
+        destination = "${NOMAD_SECRETS_DIR}/hcloud-token.env"
+        env         = true
+      }
+
+      csi_plugin {
+        id        = "csi.hetzner.cloud"
+        type      = "node"
+        mount_dir = "/csi"
+      }
+
+      resources {
+        cpu    = 100
+        memory = 64
+      }
+    }
+  }
+}
+```
+
+5. Deploy the Jobs:
+
+The following commands deploy the job resources created previously on your Nomad cluster:
+
+```sh
+nomad job run hcloud-csi-controller.hcl
+nomad job run hcloud-csi-node.hcl
+```
+
+6. Verify the status:
+
+To ensure the plugin is running and healthy, check the web UI on path `/ui/csi/plugins/csi.hetzner.cloud` or by using the CLI:
+
+```sh
+nomad plugin status
+```
+
+### Volumes Setup
+
+1. Define a Volume:
+
+Create a file `db-vol.hcl` for the volume resource:
+
+```
+# file: db-vol.hcl
+
+type      = "csi"
+id        = "db-vol"
+name      = "db-vol"
+namespace = "default"
+plugin_id = "csi.hetzner.cloud"
+
+capability {
+  access_mode     = "single-node-writer"
+  attachment_mode = "file-system"
+}
+
+mount_options {
+  fs_type     = "ext4"
+  mount_flags = ["discard", "defaults"]
+}
```

Comment:
Can we add a note here that this will create the volume in the Hetzner Cloud Location where the server with the controller is running?

If another location is required, users will need to specify [topology_request](https://developer.hashicorp.com/nomad/docs/other-specifications/volume/topology_request) (not tested):

```hcl
topology_request {
  required {
   # Use your desired location name here
    topology { segments { "csi.hetzner.cloud/location" = "fsn1" } } 
  }
}
```

Reply:
Sure, let's add a note, I am also okay to add a topology request example (never had that use-case though), we should problaby also add a link to https://developer.hashicorp.com/nomad/docs/other-specifications/volume so users find specific docs to volume handling in nomad quicker.